src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -1565,6 +1564,80 @@ struct WindowFunctionLagLeadInFrame final : public WindowFunction
     }
 };
 
+struct WindowFunctionNthValue final : public WindowFunction
+{
+    WindowFunctionNthValue(const std::string & name_,
+    const DataTypes & argument_types_, const Array & parameters_)
+    : WindowFunction(name_, argument_types_, parameters_)
+    {
+        if (!parameters.empty())
+        {
+            throw Exception(ErrorCodes::BAD_ARGUMENTS,
+                "Function {} cannot be parameterized", name_);
+        }
+
+        if (argument_types.empty())
+        {
+            throw Exception(ErrorCodes::BAD_ARGUMENTS,
+                "Function {} takes at least one argument", name_);
+        }
+
```

Comment:
If it always needs two arguments, we probably don't need this check, the next one is enough.